packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx

```diff
@@ -334,123 +247,71 @@ export function skipDataTransport<T extends Record<string, any>>(
+
+    const [controller, eventStream] = getInjectableEventStream();
+
+    const streamObservable = new Observable<
+      Exclude<QueryEvent, { type: "started" }>
+    >((subscriber) => {
+      function consume(
+        event: ReadableStreamReadResult<ReadableStreamLinkEvent>
+      ) {
+        const value = event.value;
+        if (value) {
+          subscriber.next({ ...value, id });
+        }
+        if (event.done) {
+          subscriber.complete();
+        } else {
+          reader.read().then(consume);
+        }
+      }
+      const reader = eventStream.getReader();
+      reader.read().then(consume);
+    });
+
+    this.watchQueryQueue.push({
+      event: {
+        type: "started",
+        options: serializeOptions(options),
+        id,
+      },
+      observable: streamObservable,
+    });
+
+    return controller;
+  }
+
   watchQuery<
     T = any,
     TVariables extends OperationVariables = OperationVariables,
   >(options: WatchQueryOptions<TVariables, T>) {
-    const { cacheKeyArr } = this.identifyUniqueQuery(options);
-
     if (
-      options.fetchPolicy !== "cache-only" &&
-      options.fetchPolicy !== "standby" &&
-      !(options.context as InternalContext | undefined)?.[
-        skipDataTransportKey
-      ] &&
-      !this.forwardedQueries.peekArray(cacheKeyArr)
+      !(options.context as InternalContext | undefined)?.[skipDataTransportKey]
     ) {
-      // don't transport the same query over twice
-      this.forwardedQueries.lookupArray(cacheKeyArr);
-      const observableQuery = super.watchQuery(options);
-      const queryInfo = observableQuery["queryInfo"] as QueryInfo;
-      const id = queryInfo.queryId as TransportIdentifier;
-
-      const streamObservable = new Observable<
-        Exclude<QueryEvent, { type: "started" }>
-      >((subscriber) => {
-        const { markResult, markError, markReady } = queryInfo;
-        queryInfo.markResult = function (result: FetchResult<any>) {
-          subscriber.next({
-            type: "data",
-            id,
-            result,
-          });
-          return markResult.apply(queryInfo, arguments as any);
-        };
-        queryInfo.markError = function () {
-          subscriber.next({
-            type: "error",
-            id,
-          });
-          subscriber.complete();
-          return markError.apply(queryInfo, arguments as any);
-        };
-        queryInfo.markReady = function () {
-          subscriber.next({
-            type: "complete",
-            id,
-          });
-          subscriber.complete();
-          return markReady.apply(queryInfo, arguments as any);
-        };
-      });
-
-      this.watchQueryQueue.push({
-        event: {
-          type: "started",
-          options: serializeOptions(options),
-          id,
-        },
-        observable: streamObservable,
+      return super.watchQuery({
+        ...options,
+        context: teeToReadableStream(() => this.pushEventStream(options), {
```

Comment:
Moving this into a callback that only executes if the link is actually called means we will never inject a "started" event (or worse, a stream) when the query is deduplicated or for some other reason doesn't cause a network request.